src/cclib/io/ccio.py

```diff
@@ -355,29 +356,33 @@ def _determine_output_format(outputtype, outputdest):
     #  1. outputtype
     #  2. outputdest
 
+    outputclasses = {
+        'cjson': cjsonwriter.CJSON,
+        'json': cjsonwriter.CJSON,
+        'cml': cmlwriter.CML,
+        'xyz': xyzwriter.XYZ,
+        'molden': moldenwriter.MOLDEN
+    }
+
     # First check outputtype.
     if isinstance(outputtype, str):
-        if outputtype.lower() in ('cjson', 'json'):
-            outputclass = cjsonwriter.CJSON
-        elif outputtype.lower() == 'cml':
-            outputclass = cmlwriter.CML
-        elif outputtype.lower() == 'xyz':
-            outputclass = xyzwriter.XYZ
+        try:
+            outputclass = outputclasses[outputtype.lower()]
+        except:
+            # Determine the outputclass using outputdest in else part.
+            pass
```

Comment:
KeyError with a message is being raised, and the case is being covered in the unit test.